src/data/transforms/mod.rs

```diff
@@ -21,6 +21,8 @@ pub use self::standardize::Standardizer;
 
 /// Trait for data transformers
 pub trait Transformer<T> {
+    /// Fit Transformer to input data, and stores the transformation in the Transformer
+    fn fit(&mut self, inputs: &T) -> Result<(), error::Error>;
     /// Transforms the inputs and stores the transformation in the Transformer
```

Comment:
A point is whether calling ``transform`` before ``fit`` is allowed  as standard behavior.

I'm + 1 to remove it in future. thus better to show warning and do not describe the behavior in the doc?